controller/Delivery.php

```diff
@@ -0,0 +1,348 @@
+            \common_Logger::w('No delivery service defined for proctoring');
+            $this->returnError('Proctoring interface not available');
+        }
+
+    }
+
+    /**
+     * Gets the list of test takers assigned to all deliveries
+     *
+     * @throws \common_Exception
+     * @throws \oat\oatbox\service\ServiceNotFoundException
+     */
+    public function allDeliveriesTestTakers() {
+
+        try {
+
+            $testCenter      = $this->getCurrentTestCenter();
+            $requestOptions = $this->getRequestOptions();
+            
+            $this->returnJson(DeliveryHelper::getAllDeliveryTestTakers($testCenter, $requestOptions));
+            
+
+        } catch (ServiceNotFoundException $e) {
+            \common_Logger::w('No delivery service defined for proctoring');
+            $this->returnError('Proctoring interface not available');
+        }
+
+    }
+
+    /**
+     * Gets the list of test takers available for the proctor
+     *
+     * @throws \common_Exception
+     * @throws \oat\oatbox\service\ServiceNotFoundException
+     */
+    public function availableTestTakers() {
+
+        $deliveryId = $this->getRequestParameter('id');
+
+        try {
+
+            $requestOptions = $this->getRequestOptions();
+            $testTakers = DeliveryHelper::getAvailableTestTakers($deliveryId, $requestOptions);
+            $this->returnJson($testTakers);
+
+        } catch (ServiceNotFoundException $e) {
+            \common_Logger::w('No delivery service defined for proctoring');
+            $this->returnError('Proctoring interface not available');
+        }
+
+    }
+
+    /**
+     * Assign a test taker to a delivery
+     *
+     * @throws \common_Exception
+     * @throws \oat\oatbox\service\ServiceNotFoundException
+     */
+    public function assign() {
+
+        $deliveryId = $this->getRequestParameter('id');
+        $testTakerId = $this->getRequestParameter('tt');
+
+        if (!is_array($testTakerId)) {
+            $testTakerId = array($testTakerId);
+        }
+
+        try {
+
+            $deliveryService = $this->getServiceManager()->get('taoProctoring/delivery');
+
+            $result = true;
+            foreach($testTakerId as $ttId) {
+                $result = $deliveryService->assignTestTaker($ttId, $deliveryId) && $result;
+            }
+
+            $this->returnJson(array(
+                'success' => $result
+            ));
+
+        } catch (ServiceNotFoundException $e) {
+            \common_Logger::w('No delivery service defined for proctoring');
+            $this->returnError('Proctoring interface not available');
+        }
+
+    }
+
+    /**
+     * Authorise a test taker to run the delivery
+     *
+     * @throws \common_Exception
+     * @throws \oat\oatbox\service\ServiceNotFoundException
+     */
+    public function authorise() {
+
+        $deliveryId = $this->getRequestParameter('id');
+        $testTakerId = $this->getRequestParameter('tt');
```

Comment:
Woops, this is old code just taken from the previous version. It will be change soon, as we will work on this aspect during this sprint